compiler/src/test/java/com/squareup/anvil/compiler/codegen/ksp/SimpleSymbolProcessor.kt

```diff
@@ -0,0 +1,40 @@
+package com.squareup.anvil.compiler.codegen.ksp
+
+import com.google.devtools.ksp.processing.Dependencies
+import com.google.devtools.ksp.processing.Resolver
+import com.google.devtools.ksp.processing.SymbolProcessorEnvironment
+import com.google.devtools.ksp.processing.SymbolProcessorProvider
+import com.google.devtools.ksp.symbol.KSAnnotated
+import com.squareup.anvil.compiler.api.AnvilApplicabilityChecker
+import com.squareup.anvil.compiler.internal.testing.parseSimpleFileContents
+import java.io.OutputStreamWriter
+import java.nio.charset.StandardCharsets
+
+internal fun simpleSymbolProcessor(
+  mapper: AnvilSymbolProcessor.(resolver: Resolver) -> List<String>
+): SymbolProcessorProvider = AnvilSymbolProcessorProvider(
+  AnvilApplicabilityChecker.always()
+) { env -> SimpleSymbolProcessor(env, mapper) }
+
+private class SimpleSymbolProcessor(
+  override val env: SymbolProcessorEnvironment,
+  private val mapper: AnvilSymbolProcessor.(resolver: Resolver) -> List<String>
+) : AnvilSymbolProcessor() {
+  override fun processChecked(resolver: Resolver): List<KSAnnotated> {
+    this.mapper(resolver)
+      .map { content ->
+        val (packageName, fileName) = parseSimpleFileContents(content)
+
+        val dependencies = Dependencies(aggregating = false, sources = emptyArray())
+        val file = env.codeGenerator.createNewFile(dependencies, packageName, "$fileName.kt")
+        // Don't use writeTo(file) because that tries to handle directories under the hood
```

Comment:
What ends up happening if `writeTo` was used here? Would it result in an obvious failure or something more cryptic/delayed? Thinking about if this is a footgun that could bite us elsewhere in the future

Reply:
Under the hood Kotlinpoet tries to make directories for package segments, but KSP does that for us already so we don't want to double-nest. It wouldn't obviously fail, it would just create a duplicate bested package structure. I don't think it's something we need to really guard against as this is the only place we do this custom.